### ISSUE TITLE
Battery bank charging with powersaver

### ISSUE THREAD

**Issue by torsteinelv** [ottopaulsen/node-red-contrib-power-saver#108]
Hello. 

I have been charging my batteries for a couple of days with the powersaver to reuse power when its expensive, and the savings are big every day.

This is my plan for tomorrow:
![image](https://user-images.githubusercontent.com/18403384/199975300-1698aa1d-d404-418f-bbf8-e7478f661d4d.png)

Wheen you look at 10pm to 00am i think its just stupid to charge the batteries caused of when reusing power is cheaper. Is there a way to stop this? Better for me to just keep the battery empty when there is nothing to save.

**Comment by ottopaulsen** [comment]
Looks like you are using the Best Save node, is that correct?
The Lowest Price node is better suited for charging batteries.

**Comment by torsteinelv** [comment]
sorry forgot to mention that i already use lowest price node:
![image](https://user-images.githubusercontent.com/18403384/199984699-606ccd2e-300a-4007-bcd1-c6511590a388.png)
![image](https://user-images.githubusercontent.com/18403384/199984819-e3625204-7cd9-41f2-894b-b539c552022c.png)

**Comment by ottopaulsen** [comment]
Then you should fix the graph so that the price is shown correctly. It seems to be smoothed out, and that makes it hard to see what the price actually is.

**Comment by ottopaulsen** [comment]
But, hey, I think I see your issue. The period is set from 00:00 to 00:00. The last 2 hours on the 4th is actually among the 8 cheapest that day. The next day is another calculation.

You can change to use time from 16:00 to 08:00 or something like that.

**Comment by torsteinelv** [comment]
Ok. But then no charge between 8 and 16 is done?

**Comment by torsteinelv** [comment]
![Screenshot_20221104-163252_Home Assistant](https://user-images.githubusercontent.com/18403384/200014990-b93ca59f-1a4a-497c-9614-a99a5fa11df9.jpg)
Much better now. But missing the feature to not have to define how many hours on

**Comment by kristianolofsson** [comment]
Hi, I’m also looking for this. Possible to share your flow?
I’m using a 15kwh battery and would like to calculate a predicted forcaste if I should charge/discharge or just use the grid (to save the battery) with every hour.

**Comment by torsteinelv** [comment]
Yes. I will add forecast also. I can svare the flow but im not happy with it yet

**Comment by torsteinelv** [comment]
`[
   {
      "id":"96227012d2cecae2",
      "type":"mqtt out",
      "z":"8b62262273fd21db",
      "name":"test2",
      "topic":"",
      "qos":"0",
      "retain":"false",
      "respTopic":"",
      "contentType":"",
      "userProps":"",
      "correl":"",
      "expiry":"",
      "broker":"24fbcfb5.569ea",
      "x":410,
      "y":1060,
      "wires":[
         
      ]
   },
   {
      "id":"26eaf7ebbbefe1d1",
      "type":"function",
      "z":"8b62262273fd21db",
      "name":"PÅ",
      "func":"msg.payload = 57.2;\nmsg.topic = \"solar_assistant/inverter_1/voltage_point_1/set\";\nreturn msg;",
      "outputs":1,
      "noerr":0,
      "initialize":"",
      "finalize":"",
      "libs":[
         
      ],
      "x":250,
      "y":880,
      "wires":[
         [
            "96227012d2cecae2"
         ]
      ]
   },
   {
      "id":"55fcc3c6d71df581",
      "type":"function",
      "z":"8b62262273fd21db",
      "name":"AV",
      "func":"msg.payload = 47;\nmsg.topic = \"solar_assistant/inverter_1/voltage_point_1/set\";\nreturn msg;",
      "outputs":1,
      "noerr":0,
      "initialize":"",
      "finalize":"",
      "libs":[
         
      ],
      "x":250,
      "y":920,
      "wires":[
         [
            "96227012d2cecae2"
         ]
      ]
   },
   {
      "id":"24fbcfb5.569ea",
      "type":"mqtt-broker",
      "name":"MQTT",
      "broker":"10.10.10.101",
      "port":"1883",
      "clientid":"",
      "autoConnect":true,
      "usetls":false,
      "compatmode":false,
      "protocolVersion":"4",
      "keepalive":"60",
      "cleansession":true,
      "birthTopic":"",
      "birthQos":"0",
      "birthPayload":"",
      "birthMsg":{
         
      },
      "closeTopic":"",
      "closeQos":"0",
      "closePayload":"",
      "closeMsg":{
         
      },
      "willTopic":"",
      "willQos":"0",
      "willPayload":"",
      "willMsg":{
         
      },
      "userProps":"",
      "sessionExpiry":""
   }
]`

**Comment by kristianolofsson** [comment]
Thanks! Yes a forecast would be nice. The question is to how to push the forecast values to HA.

- Read the nord pool values and use the best value node to get hold of the “best” hours to charge/discharge. 
- Show the forecast result in a graph. Eg using apex threshold colors to tell if the nordpool price is with level :“cheap-Charge battery”, “Normal-use grid”, “Expensive-Discharge battery”.
- The result should also every hour tell the actual level to act on.

**Comment by torsteinelv** [comment]
Tale consumption from history and calculate how much energy needed until forcast energy so not charging to much from grid. Also plan for appliances to me using solar or buy when its cheapest.

**Comment by torsteinelv** [comment]
Is it possible to change the best save to change places to the lowest hours only? This would work much better when able to define the savings.

**Comment by ottopaulsen** [comment]
I am working on a node that can combine schedules from multiple strategy nodes. With this you can combine a Best Save and a Lowest Price with AND to get only the Best Save hours that are among the x cheapest hours. Not sure it will give you exactly what you want, but maybe. It will be released in v4.0.0, hopefully in a few days or weeks.

**Comment by ottopaulsen** [comment]
Please take a look at the new [Schedule Merger](https://powersaver.no/nodes/ps-schedule-merger.html#description) node in version 4.

**Comment by torsteinelv** [comment]
Thanks. Trying this out now. What about adding a schedule from forcast pv energy to add to the prices? For people with solar powered homes a forcast energy for the next day can be planned into the schedule and eq hotwater can be turned on here?

I Can try to make this myself just modifying the flow but I think many people would like this feature.
